### PR TITLE
[SPARK-41293][SQL][TESTS] Code cleanup for `assertXXX` methods in `ExpressionTypeCheckingSuite`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr do some code clean up for `assertXXX` method in `ExpressionTypeCheckingSuite`:

1.  Reuse `analysisException` instead of duplicate `intercept[AnalysisException](assertSuccess(expr))` in `assertErrorForXXX` methods.
2. remove  `assertError` method that is no longer used 
3. Change `assertErrorForXXX` methods access scope to `private` due to they are only used in `ExpressionTypeCheckingSuite`.

### Why are the changes needed?
Code clean up.

### Does this PR introduce _any_ user-facing change?
No, just for test

### How was this patch tested?
Pass GitHub Actions

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @MaxGekk

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ExpressionTypeCheckingSuite.scala [hunk 1]
```diff
@@ -44,67 +44,43 @@ class ExpressionTypeCheckingSuite extends SparkFunSuite with SQLHelper with Quer
...
-    }
-    assert(e.getMessage.contains(
-      s"cannot resolve '${expr.sql}' due to data type mismatch:"))
-    assert(e.getMessage.contains(errorMessage))
+  private def assertError(
+      expr: Expression,
+      errorClass: String,
+      messageParameters: Map[String, String]): Unit = {
+    checkError(
+      exception = analysisException(expr),
```
Comment (LuciferYang) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ExpressionTypeCheckingSuite.scala:52]:
``` scala
private def analysisException(expr: Expression): AnalysisException = {
    intercept[AnalysisException](assertSuccess(expr))
  }
```

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master. I have checked the test suite locally.
Thank you, @LuciferYang.

**Comment by LuciferYang** [comment]
thanks @MaxGekk

**Pull request closed without merging**